Instructions on handing interactive ISO Install and where to locate grub.cfg (#298)
harvester/docs: docs/install/usb-install.md

# sudo dd if=<path_to_iso> of=<path_to_usb_device> bs=64k
```

> Sometimes using a USB to mount the Harvester ISO disk directs you to Grub2. In this case, it helps to know what to set the root partition to and the location of `grub.cfg`.

Reviewer: I think this should move to `Common issues`?

You can see the current root partition in the `grub.cfg`. To view the `grub.cfg` from the ISO, run the command `chainloader /boot/grub2/x86_64-efi/grub.efi`.

> To set a new root partition, run the command `set root = (hdN,msdos1)`, where N denotes N numbers of possible drives. N should be the drive your USB is mounted to.

Reviewer 2: the command should read:
`set root=(hdN,msdos1)` 

as if it has " " (spaces) in the command, the command will fail if I remember correctly :thinking: :smile:

## Common issues

###

Sometimes using a USB to mount the Harvester ISO disk directs you to Grub2. In this case, it helps to know how to set the root partition and where to locate `grub.cfg`.

> You can see the current root partition in the `grub.cfg`. To view the `grub.cfg` from the ISO, run the command `chainloader /boot/grub2/x86_64-efi/grub.efi`.

Reviewer 2: ```suggestion
You can see the current root partition in the `grub.cfg`, once you've set the root to be within where the ISO lives on the USB. To view the `grub.cfg` from the ISO, run the command `cat /boot/grub2/grub.cfg`.
```

So to view it, the user will need to look at the `grub.cfg` with something like `cat` - sadly at least on our Grub 2.06, it seems `less` isn't an available command :sweat_smile:  - but! - `cat` should work great for viewing.

To set a new root partition, run the command `set root=(hdN,msdos1)`, where N denotes N numbers of possible drives. N should be the drive your USB is mounted to.

> 

Author: @irishgordo In [issue#3590](https://github.com/harvester/harvester/issues/3590), you mentioned "Then they'd be able to fire off the boot command from Grub2 and be taken to the interactive ISO installation." Should that step be added here as well?

Reviewer 2: Ahh I see what you're saying :smile:  - I've added a suggestion below :+1:

> 

Reviewer 2: ```suggestion

To launch into the GNU GRUB2 Boot Menu from the ISO, once the root has been set to where the ISO is located at the partition of `msdos1` - you can run `chainloader /boot/grub2/x86_64-efi/grub.efi` to get loaded into the GRUB2 Boot Menu for the Harvester Installer.
```

## Common issues

###

> Sometimes using a USB to mount the Harvester ISO disk directs you to Grub2. In this case, it helps to know how to set the root partition and where to locate `grub.cfg`.

Reviewer: `directs you to GRUB2` is a bit confusing, if the problem is dropping into GRUB2 shell directly when booting up maybe we can describe it more clearly? (if I'm understanding this correctly)

## Known issues

###

> Sometimes using a USB to mount the Harvester ISO disk directs you to GRUB2. In this case, it helps to know how to set the root partition and where to locate `grub.cfg`.

Author: ```suggestion
Sometimes when using a USB to mount the Harvester ISO disk, it drops you in the GRUB2 shell while booting. In this case, it helps to know how to locate `grub.cfg`  and set the root partition to proceed with the installation.
```

> You can see the current root partition in the `grub.cfg`, once you've set the root to be within where the ISO lives on the USB. To view the `grub.cfg` from the ISO, run the command `cat /boot/grub2/grub.cfg`.

Author: ```suggestion
To view the `grub.cfg` from the ISO, run the command `cat /boot/grub2/grub.cfg`. You can see the current root partition in the `grub.cfg`.
```

To set a new root partition, run the command `set root=(hdN,msdos1)`, where N denotes N numbers of possible drives. N should be the drive your USB is mounted to.

> To launch into the GNU GRUB2 Boot Menu from the ISO, once the root has been set to where the ISO is located at the partition of `msdos1` - you can run `chainloader /boot/grub2/x86_64-efi/grub.efi` to get loaded into the GRUB2 Boot Menu for the Harvester Installer.

Author: ```suggestion
Once the root partition has been set, where the ISO is located in the `msdos1` partition, run the command `chainloader /boot/grub2/x86_64-efi/grub.efi` to get dropped into the GRUB2 Boot Menu for the Harvester Installer.
```
